src/hinge.cpp

```diff
@@ -0,0 +1,125 @@
+#include "wayfire/plugin.hpp"
+#include "wayfire/signal-definitions.hpp"
+#include "wayfire/input-device.hpp"
+#include <wayfire/util/log.hpp>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <unistd.h>
+#include <thread>
+
+class wayfire_hinge : public wf::plugin_interface_t
+{
+    enum thread_message {
```

Comment:
If `thread_message` is supposed to fit into a `char`, then you should declare the enum like so:
```c++
enum thread_message : char { ... }
```

then later you may be able to remove the char cast in send_message() since this will be implicitly done.